src/main/java/com/example/travelservice/services/LocationServiceImpl.java

```diff
@@ -29,8 +32,28 @@ public Location createLocation(Location location) {
     }
 
     @Override
-    public Location findById(Long id) {
-        return locationRepository.findById(id).orElseThrow();
+    public Location findById(Long locationId) {
+        return locationRepository.findById(locationId).orElseThrow(() ->
+                new NotFoundException("Location with id" + locationId + " not found"));
+    }
+
+    @Override
+    public Location updateCity(Long locationId, String cityName) {
+        Location location = locationRepository.findById(locationId).orElseThrow(() ->
+                new NotFoundException("Location with id" + locationId + " not found"));
+        location.setCityName(cityName);
+        return locationRepository.save(location);
+    }
+
+    @Override
+    public Location updateLocation(Long locationId, LocationDto locationDto) {
+        Location location = locationConverter.convert(locationDto);
+        location.setId(locationId);
+        if(!locationNotExists(location)){
+            return locationRepository.save(location);
+        } else{
+            return createLocation(location);
```

Comment:
if location by such id doesn't exist you have to throw an exception, because user is trying to update non-existing entity